File: myblog-server/src/main/java/com/site/blog/model/dto/AjaxPutPage.java
package com.site.blog.model.dto;

import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import lombok.Data;
import lombok.ToString;


@Data

public class AjaxPutPage<T> {
    /**
     * 当前页码
     */
    Integer page;

    /**
     * 每页显示
     */
    Integer limit;

    /**
     * 从多少开始
     */
    Integer start;
    Boolean show;
    /**
     * 删除
     */
    Boolean deleted;
    /**
     * 条件类
     */
    T condition;


    /**
     * 将符合Layui的格式转成mybtais-plus分页的page
     *
     * @return
     */
    public Page<T> putPageToPage() {
        return new Page<>(this.page, this.limit);
    }


}

File: myblog-kt/src/main/kotlin/com/site/blog/model/dto/AjaxPutPage.kt
package com.site.blog.model.dto

import com.baomidou.mybatisplus.extension.plugins.pagination.Page

data class AjaxPutPage<T> (
    /**
     * 当前页码
     */
    var page: Int? = null,

    /**
     * 每页显示
     */
    var limit: Int? = null,

    /**
     * 从多少开始
     */
    var start: Int? = null,
    var show: Boolean? = null,

    /**
     * 删除
     */
    var deleted: Boolean? = null,

    /**
     * 条件类
     */
    var condition: T? = null,



){
    /**
     * 将符合Layui的格式转成mybtais-plus分页的page
     *
     * @return
     */
    fun putPageToPage(): Page<T> {
        return Page(page!!.toLong(), limit!!.toLong())
    }
}